autoextract_poet/items.py

```diff
@@ -124,3 +136,63 @@ def from_dict(cls, item: Optional[Dict]):
         ))
 
         return super().from_dict(new_item)
+
+
+@attr.s(auto_attribs=True, slots=True)
+class ProductFromList(Item):
+
+    probability: Optional[float] = None
+    url: Optional[str] = None
+    name: Optional[str] = None
+    offers: List[Offer] = attr.Factory(list)
+    sku: Optional[str] = None
+    brand: Optional[str] = None
+    mainImage: Optional[str] = None
+    images: List[str] = attr.Factory(list)
+    description: Optional[str] = None
+    aggregateRating: Optional[Rating] = None
+
+    @classmethod
+    def from_dict(cls, item: Optional[Dict]):
+        if not item:
+            return None
+
+        new_item = dict(**item)
+        new_item.update(dict(
+            aggregateRating=Rating.from_dict(item.get("aggregateRating")),
+            offers=Offer.from_list(item.get("offers", [])),
+        ))
+
+        return super().from_dict(new_item)
+
+
+@attr.s(auto_attribs=True, slots=True)
+class PaginationLink(Item):
+
+    url: Optional[str] = None
+    text: Optional[str] = None
+
+
+@attr.s(auto_attribs=True, slots=True)
+class ProductList(Item):
+
+    url: Optional[str] = None
+    products: List[ProductFromList] = attr.Factory(list)
+    breadcrumbs: List[Breadcrumb] = attr.Factory(list)
+    paginationNext: Optional[PaginationLink] = None
+    paginationPrevious: Optional[PaginationLink] = None
+
+    @classmethod
+    def from_dict(cls, item: Optional[Dict]):
+        if not item:
+            return None
+
+        new_item = dict(**item)
+        new_item.update(dict(
+            products=ProductFromList.from_list(item.get("products", [])),
+            breadcrumbs=Breadcrumb.from_list(item.get("breadcrumbs", [])),
+            paginationNext=PaginationLink.from_dict(item.get("paginationNext", {})),
```

Comment:
I guess it needs formatting

Reply:
What do you mean?

Reply:
It was about the line length, I think it was a bit odd when I looked